tests/meta/transfer/test_external_data.py

```diff
@@ -174,6 +175,30 @@ def test_add_transfer_to_housekeeper(
     )
 
 
+def test_get_available_samples(
+    analysis_store_trio,
+    customer_id: str,
+    external_data_api: ExternalDataAPI,
+    sample_obj: models.Sample,
+    ticket_nr: int,
+    tmpdir_factory,
+):
+    # GIVEN that the root directory does not contain any correct folders
+    # THEN the function should return an empty list
+    transferred_samples = external_data_api.get_available_samples(
+        folder=Path("."), ticket_id=ticket_nr
+    )
+    assert transferred_samples == []
+
+    # GIVEN one such sample exists
+    tmp_dir_path: Path = Path(tmpdir_factory.mktemp(sample_obj.internal_id, numbered=False)).parent
+    ExternalDataAPI.get_destination_path.return_value = tmp_dir_path
+    transferred_samples = external_data_api.get_available_samples(
+        folder=tmp_dir_path, ticket_id=ticket_nr
+    )
+    assert transferred_samples == [sample_obj]
+
```

Comment:
Move this to a separate test function to make it easier to understand why a test fails and to keep things separate as much as possible